client/cmd/dexc/ui/depthchart.go

```diff
@@ -0,0 +1,271 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package ui
+
+import (
+	"fmt"
+	"math"
+	"sync"
+
+	"github.com/gdamore/tcell"
+	"github.com/rivo/tview"
+)
+
+var (
+	chartColor = tcell.GetColor("#0c456b")
+)
+
+type depthPoint struct {
+	x float64
+	y float64
+}
+
+// depthChart is a simple character-based chart. Right now, the chart only
+// works for a fixed chart size which cannot be changed.
+type depthChart struct {
+	*tview.Box
+	focus    bool
+	mtx      sync.RWMutex
+	seq      uint64
+	drawID   uint64
+	runeRows []string
+}
+
+func newDepthChart() *depthChart {
+	dc := &depthChart{
+		Box: tview.NewBox(),
+		seq: 1,
+	}
+	dc.SetBorder(true)
+	dc.refresh()
+	return dc
+}
+
+// Need to handle concurrency for chart updates.
+func (c *depthChart) newScreenRows() []string {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	if !c.focus {
+		return nil
+	}
+	if c.seq == c.drawID {
+		// Nothing to redraw.
+		return nil
+	}
+	c.refresh()
+	c.seq = c.drawID
+	return c.runeRows
+}
+
+// Draw hijacks the embedded Box method to draw on the screen.
+func (c *depthChart) Draw(screen tcell.Screen) {
+	screenX, screenY, width, _ := c.GetRect()
+	rows := c.newScreenRows()
+	for y, row := range rows {
+		tview.Print(screen, row, 0+screenX, y+screenY, width, tview.AlignLeft, chartColor)
+	}
+}
+
+// refresh recalculates the chart rune rows.
+func (c *depthChart) refresh() {
+	rotations := float64(2.5)
+	numPts := 400
+	amp := float64(5)
+	pts := make([]depthPoint, 0, numPts)
+	for i := 0; i < numPts; i++ {
+		x := float64(i) / float64(numPts) * rotations * 2 * math.Pi
+		pts = append(pts, depthPoint{
+			x: float64(i),
+			y: math.Sin(x) * amp,
+		})
+	}
+	width, height := marketChartWidth, marketChartHeight
+	if width == 0 || height == 0 {
+		return
+	}
+	runeRows := make([]string, 0, height)
+	edge, err := interpolate(pts, width, height)
+	if err != nil {
+		// Don't update the UI from a Draw method.
+		return
+	}
+
+	for iy := 0; iy < height; iy++ {
+		y := height - iy - 1
+		row := make([]rune, width)
```

Comment:
Did you mean to use `make([]rune, 0, width)` with the `append`s or is the intent to prepend with a full row of zero-value runes?